### DISCUSSION TITLE
Feature Bundles & Dependency Management

### DISCUSSION THREAD

**Discussion by ignatz** [rscarson/rustyscript#300] [General]
Huge fan of rustyscript :heart:

I noticed that you added deno_telemetry in 0.11 and made it part of the "web" feature bundle, which in turn added quite a few build targets and binary size for me. Which is not the end of the world and I could probably roll my own bundle if I was to care enough. I totally understand that you cannot offer bundles for all possible combinations.

I guess, I was mostly just curious what your guiding principle is for what goes into the "web" feature and if telemetry is  a good fit? Thanks

EDIT: forgot to point to https://crates.io/crates/deno_web, which presumably was the starting point.

**Comment by rscarson** [comment]
Actually that change was because telemetry was added as a dependency of deno_fetch, one of the web bundle features 

That set was specifically chosen as the minimal set of extensions needed to bootstrap deno_web, and for now that includes telemetry.

That may change of deno's end in the future, at which point to will unroll it again

**Comment by ignatz** [comment]
Thanks! That makes perfect sense, just hoped it was a mistake. Hope the deno folks settle on a approach more akin to the "tracing" packages to get telemetry from fetch w/o depending on opentelemetry :shrug: 

Thanks!
